src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexInterpreter.cs

```diff
@@ -2,319 +2,263 @@
 
-// The RegexInterpreter executes a block of regular expression codes
-// while consuming input.
-
 using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
 using System.Globalization;
 using System.Runtime.CompilerServices;
 
 namespace System.Text.RegularExpressions
 {
+    /// <summary>Executes a block of regular expression codes while consuming input.</summary>
     internal sealed class RegexInterpreter : RegexRunner
     {
+        private const int LoopTimeoutCheckCount = 2048; // conservative value to provide reasonably-accurate timeout handling.
+
         private readonly RegexCode _code;
-        private readonly CultureInfo _culture;
+        private readonly TextInfo _textInfo;
+
         private int _operator;
         private int _codepos;
         private bool _rightToLeft;
         private bool _caseInsensitive;
-        private const int LoopTimeoutCheckCount = 2048; // A conservative value to guarantee the correct timeout handling.
 
         public RegexInterpreter(RegexCode code, CultureInfo culture)
         {
-            Debug.Assert(code != null, "code cannot be null.");
-            Debug.Assert(culture != null, "culture cannot be null.");
+            Debug.Assert(code != null, "code must not be null.");
+            Debug.Assert(culture != null, "culture must not be null.");
 
             _code = code;
-            _culture = culture;
+            _textInfo = culture.TextInfo;
         }
 
-        protected override void InitTrackCount()
-        {
-            runtrackcount = _code.TrackCount;
-        }
+        protected override void InitTrackCount() => runtrackcount = _code.TrackCount;
 
         private void Advance(int i)
         {
-            _codepos += (i + 1);
+            _codepos += i + 1;
             SetOperator(_code.Codes[_codepos]);
         }
 
         private void Goto(int newpos)
         {
-            // when branching backward, ensure storage
+            // When branching backward, ensure storage.
             if (newpos < _codepos)
+            {
                 EnsureStorage();
+            }
 
-            SetOperator(_code.Codes[newpos]);
             _codepos = newpos;
+            SetOperator(_code.Codes[newpos]);
         }
 
-        private void Textto(int newpos)
-        {
-            runtextpos = newpos;
-        }
-
-        private void Trackto(int newpos)
-        {
-            runtrackpos = runtrack!.Length - newpos;
-        }
-
-        private int Textstart()
-        {
-            return runtextstart;
-        }
-
-        private int Textpos()
-        {
-            return runtextpos;
-        }
+        private void Trackto(int newpos) => runtrackpos = runtrack!.Length - newpos;
 
-        // push onto the backtracking stack
-        private int Trackpos()
-        {
-            return runtrack!.Length - runtrackpos;
-        }
+        /// <summary>Push onto the backtracking stack.</summary>
```

Comment:
(nit) this is kind of a weird summary for this method. Was the original comment supposed to apply to the group of methods that follow?

Reply:
Thanks. Fixed to be on the next method instead.